script-runner/AutomatedTestRunner.hs

```diff
@@ -0,0 +1,400 @@
+
+maybeAddPeers :: [NodeId] -> NodeParams -> NodeParams
+maybeAddPeers [] params = params
+maybeAddPeers peers nodeParams = nodeParams { npNetworkConfig = (npNetworkConfig nodeParams) { ncTopology = TopologyAuxx peers } }
+
+addQueuePolicies :: NodeParams -> NodeParams
+addQueuePolicies nodeParams = do
+  let
+    topology = ncTopology $ npNetworkConfig nodeParams
+  nodeParams { npNetworkConfig = (npNetworkConfig nodeParams)
+    { ncEnqueuePolicy = topologyEnqueuePolicy topology
+    , ncDequeuePolicy = topologyDequeuePolicy topology
+    , ncFailurePolicy = topologyFailurePolicy topology
+    }
+  }
+
+runWithConfig :: (TestScript a, HasCompileInfo, HasConfigurations) => ScriptRunnerOptions -> InputParams a -> Genesis.Config -> WalletConfiguration -> TxpConfiguration -> NtpConfiguration -> IO ()
+runWithConfig ScriptRunnerOptions{..} inputParams genesisConfig _walletConfig txpConfig _ntpConfig = do
+  let
+    cArgs@CLI.CommonNodeArgs {..} = srCommonNodeArgs
+    nArgs = CLI.NodeArgs {behaviorConfigPath = Nothing}
+  (nodeParams', _mSscParams) <- CLI.getNodeParams loggerName cArgs nArgs (configGeneratedSecrets genesisConfig)
+  let
+    nodeParams = addQueuePolicies $ maybeAddPeers srPeers $ nodeParams'
+    epochSlots = configEpochSlots genesisConfig
+    vssSK = fromMaybe (error "no user secret given") (npUserSecret nodeParams ^. usVss)
+    sscParams = CLI.gtSscParams cArgs vssSK (npBehaviorConfig nodeParams)
+    thing1 = txpGlobalSettings genesisConfig txpConfig
+    thing2 :: ReaderT InitModeContext IO ()
+    thing2 = initNodeDBs genesisConfig
+  script <- liftIO $ withEpochSlots epochSlots genesisConfig (ipScriptGetter inputParams)
+  let
+    inputParams' = InputParams2 (ipEventChan inputParams) (ipReplyChan inputParams) script
+  bracketNodeResources genesisConfig nodeParams sscParams thing1 thing2 (thing3 genesisConfig txpConfig inputParams')
+
+thing3 :: (TestScript a, HasCompileInfo, HasConfigurations) => Config -> TxpConfiguration -> InputParams2 a -> NodeResources () -> IO ()
+thing3 genesisConfig txpConfig inputParams nr = do
+  let
+    toRealMode :: AuxxMode a -> RealMode EmptyMempoolExt a
+    toRealMode auxxAction = do
+      realModeContext <- ask
+      lift $ runReaderT auxxAction $ AuxxContext { acRealModeContext = realModeContext }
+    thing2 :: Diffusion (RealMode ()) -> RealMode EmptyMempoolExt ()
+    thing2 diffusion = toRealMode (thing5 (hoistDiffusion realModeToAuxx toRealMode diffusion))
+    thing5 :: Diffusion AuxxMode -> AuxxMode ()
+    thing5 = runNode genesisConfig txpConfig nr thing4
+    thing4 :: [ (Text, Diffusion AuxxMode -> AuxxMode ()) ]
+    thing4 = workers genesisConfig inputParams
+  runRealMode updateConfiguration genesisConfig txpConfig nr thing2
+
+workers :: (HasConfigurations, TestScript a) => Genesis.Config -> InputParams2 a -> [ (Text, Diffusion AuxxMode -> AuxxMode ()) ]
+workers genesisConfig InputParams2{ip2EventChan,ip2Script,ip2ReplyChan} =
```

Comment:
I would actually write that as:
```
workers genesisConfig (InputParams2 ip2EventChan ip2Script ip2ReplyChan) =
```
which i thing is a little less noisy. I would also tend to use shorter names for what are local variables.